### PR TITLE
Support more NPM license conventions

### PR DESCRIPTION
**Pull request by g-marconet**
- Extends the NPM license retriever to cover more cases

## Description

- Can now handle SPDX license expressions
  - Which is the currently-supported way to have multiple licenses: https://docs.npmjs.com/cli/v10/configuring-npm/package-json#license
  - We opted not to lose the AND/OR/WITH logic in this pull request by extracting all the mentioned license versions. SPDX expressions should be preserved as-is
    - Another option is to parse the expression and return all licenses mentioned, but then we'd lose the AND/OR/WITH relationship. (We are not doing this)
- Also handles more "deprecated" license cases
  - Like `license: { type: str }`
  - Like `licenses: [ { type: str } ]`

## Motivation and Context

There were common license patterns that we were not reading

## How Has This Been Tested?

Tests pass and working in dev environment

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change

## Checklist

- [x] My code follows conforms to the coding standards.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: backend/lambdas/maintenance/license_retriever/license_retriever/retrievers/npm.py [hunk 1]
```diff
@@ -38,3 +30,62 @@ def get_package_info(name: str, version: str) -> dict:
...
+
+
+def get_license(item: Union[dict, str], package_name: str) -> Optional[str]:
+    if type(item) is str:
+        return item.lower()
+    elif type(item) is dict:
+        if "type" in item and type(item["type"]) is str:
+            return item["type"].lower()
+
+    LOG.warning('Could not parse a license for "%s". It may be unconventional.', package_name)
```
Comment (g-marconet) [inline backend/lambdas/maintenance/license_retriever/license_retriever/retrievers/npm.py:90]: Is warning the right log level for this?

Essentially, I wanted some log to point out if conventions change or if there is a common convention we miss. However, this input is untrusted, so an NPM package messing up the convention or similar will also trigger this path.

File: backend/lambdas/maintenance/license_retriever/license_retriever/retrievers/npm.py [hunk 2]
```diff
@@ -38,3 +30,62 @@ def get_package_info(name: str, version: str) -> dict:
...
+    #     { type: "GPL-3.0", url: "..." },
+    #     { type: "MIT", url: "..." }
+    # ]
+
+    if "license" in package_info:
+        license = package_info["license"]
+    elif "licenses" in package_info:
+        license = package_info["licenses"]
+    else:
+        LOG.debug('No license information in package info for "%s"', package_name)
```
Comment (g-marconet) [inline backend/lambdas/maintenance/license_retriever/license_retriever/retrievers/npm.py:61]: For reviewer, should this be debug? We log as "Error" on L31 when we can't find the package info.

File: backend/lambdas/maintenance/license_retriever/license_retriever/retrievers/npm.py (hunk 1 above)
Comment (Kamsiy) [reply backend/lambdas/maintenance/license_retriever/license_retriever/retrievers/npm.py:90]: A warning is perfect here

File: backend/lambdas/maintenance/license_retriever/license_retriever/retrievers/npm.py (hunk 2 above)
Comment (Kamsiy) [reply backend/lambdas/maintenance/license_retriever/license_retriever/retrievers/npm.py:61]: I think the information logged here should be a `warning` . But it also depends on how common this case is. 
If its really common, it makes sense to have it a `debug` log. If its not, then it should be a `warning`

**Review by Kamsiy — APPROVED** [review]

**Pull request merged**